### PR TITLE
Docs improvements

### PR DESCRIPTION
**Pull request by umairidris**

### CODE REVIEW

File: docs/tfengine/README.md [hunk 1]
```diff
@@ -6,9 +6,19 @@ The Terraform Engine is a tool to generate complete end-to-end Terraform
...
+
+## Terminology
+
+The Terraform engine introduces the concept of "templates". Templates can be
+used to generate
+[Terraform root modules](https://www.terraform.io/docs/glossary.html#root-module)
+specific to your Google Cloud organization and structure.
+
+Users typically pick from a set of
+[recipes](../../templates/tfengine/recipes) which implement a template for one
```
Comment (MartinPetkov) [inline docs/tfengine/README.md:19]: I think recipes need to be described differently, they're mainly a way to group components together.

File: docs/tfengine/README.md [hunk 2]
```diff
@@ -19,14 +29,69 @@ that implement these steps for you, you can quickly set up a secure and complian
...
+    (for 1 year) and to GCS (for 7 years). These configurations help align with
+    [HIPAA audit log](https://www.securitymetrics.com/blog/what-are-hipaa-compliant-system-logs)
+    requirements.
+
+- Reduce human access to the org infrastructure. Promote coding and version
+  control best practices.
+
+  - The [CICD recipe](./recipes/cicd.md) sets up a pipeline that is
+    run by Cloud Build service accounts. Through integration with Github,
+    changes to infrastructure can be made via pull requests. The hooks we set up\
```
Comment (MartinPetkov) [inline docs/tfengine/README.md:66]: Typo `\` at the end?

File: docs/tfengine/README.md [hunk 3]
```diff
@@ -19,14 +29,69 @@ that implement these steps for you, you can quickly set up a secure and complian
...
+- Allow logical folders within your hierarchy to be managed by independently,
+  thus reducing org-wide broad access to single service account and chances of
+  cascading errors.
+
+  - The [devops recipe](./recipes/devops.md) can be used on different folders
+    to setup a separate CICD pipeline and service account to manage projects
+    and resources within the folder. The service accounts of other CICD
+    pipelines cannot access these projects.
+
+- Define many security sensitive such as
```
Comment (MartinPetkov) [inline docs/tfengine/README.md:80]: The phrase "many security sensitive" seems like it's missing a word.

File: docs/tfengine/README.md (hunk 1 above)
Comment (umairidris) [reply docs/tfengine/README.md:19]: Users don't really need to know about components I feel, that is more of an implementation detail. They are using the recipes.

In our case the users pick recipes like audit.hcl, project.hcl, etc which are implementing one core piece of GCP infra.

File: docs/tfengine/README.md [hunk 4]
```diff
@@ -19,14 +29,69 @@ that implement these steps for you, you can quickly set up a secure and complian
...
+This tool helps you follow Google Cloud and Terraform best practices:
 
-- Google Cloud best practices through the use of modules from the
-[Cloud Foundation Toolkit](https://cloud.google.com/foundation-toolkit).
+- Clearly define your
+  [resource hierarchy](https://cloud.google.com/docs/enterprise/best-practices-for-enterprise-organizations#define-hierarchy)
+  in infra-as-code.
 
-- [Terraform best practices](https://www.hashicorp.com/resources/evolving-infrastructure-terraform-opencredo).
+  - The [org foundation example](../../examples/tfengine/org_foundation.hcl)
```
Comment (xingao267) [inline docs/tfengine/README.md:38]: For example, The [org foundation ...

Comment (umairidris) [reply docs/tfengine/README.md]: Done

File: docs/tfengine/README.md (hunk 1 above)
Comment (MartinPetkov) [reply docs/tfengine/README.md:19]: Up to you then, I approved already. We can change it if someone opens an issue that it's unclear what recipes are or how they work.

File: templates/tfengine/recipes/README.md [hunk 5]
```diff
@@ -1,18 +1,17 @@
 # Terraform Engine Recipes
 
+See markdown schemas for recipes [here](../../docs/tfengine/recipes).
```
Comment (xingao267) [inline templates/tfengine/recipes/README.md:3]: ../../../docs/tfengine/recipes

**Pull request merged**